### PR TITLE
[Excel] (Custom functions) Error handling

### PR DESCRIPTION
**Pull request by mscharlock**
Separating out error handling into it's own article. Content should be the same as it was!

### CODE REVIEW

**Comment by o365devx** [comment]
OPS Build status updates of commit _[adf7b70](https://github.com/OfficeDev/office-js-docs-pr/pull/778/commits/adf7b700888945fc78c8aced30e0937e6cb8f638)_: 

:clock10: **Preparing**: average preparing time is _44 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[adf7b70](https://github.com/OfficeDev/office-js-docs-pr/pull/778/commits/adf7b700888945fc78c8aced30e0937e6cb8f638)_: 

:clock10: **Incremental building**: average incremental building time is _29 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[adf7b70](https://github.com/OfficeDev/office-js-docs-pr/pull/778/commits/adf7b700888945fc78c8aced30e0937e6cb8f638)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/excel/custom-functions-best-practices.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-errors.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/excel/custom-functions-errors.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C2%5C22%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201902221855156354-778%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=8gsZaEGbgtEE4CqYgEZWJQNzfh2RNue4E%2FGbUq2wCoY%3D&st=2019-02-22T18%3A51%3A36Z&se=2019-03-25T18%3A56%3A36Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by mscharlock** [comment]
@davidchesnut do we need to make changes on this one? I remember seeing some ideas floating around in email but essentially this is just a section that separates out content, so wanted to check. Thanks!

File: docs/excel/custom-functions-errors.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Error handling within custom functions
+
+When you build an add-in that defines custom functions, be sure to include error handling logic to account for runtime errors. Error handling for custom functions is the same as [error handling for the Excel JavaScript API at large](excel-add-ins-error-handling.md).
+
+[!include[Excel custom functions note](../includes/excel-custom-functions-note.md)]
+
+In the following code sample, `.catch` will handle any errors that occur previously in the code.
+
+```js
+function getComment(x) {
```
[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/excel/custom-functions-errors.md]:
```suggestion
function getComment(commentID) {
```

File: docs/excel/custom-functions-errors.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+title: Error handling for custom functions in Excel (preview)
+localization_priority: Priority
+---
+# Error handling within custom functions
+
+When you build an add-in that defines custom functions, be sure to include error handling logic to account for runtime errors. Error handling for custom functions is the same as [error handling for the Excel JavaScript API at large](excel-add-ins-error-handling.md).
+
+[!include[Excel custom functions note](../includes/excel-custom-functions-note.md)]
+
+In the following code sample, `.catch` will handle any errors that occur previously in the code.
```
[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/excel/custom-functions-errors.md]: Ah ok - you are wondering if it's possible to tell Excel which error code to throw (e.g. you could automatically return #N/A or #GETTING_DATA in some situations?). I thought that's what we aren't able to do yet (which is why they invented #BUSY!)? I guess the alternative is that you could pass your own string along...? Maybe I'm misunderstanding your suggestion.

**Comment by o365devx** [comment]
OPS Build status updates of commit _[5eb63fe](https://github.com/OfficeDev/office-js-docs-pr/pull/778/commits/5eb63fee328426e5a647725d346edd3e3bd0e84f)_: 

:clock10: **Preparing**: average preparing time is _44 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[5eb63fe](https://github.com/OfficeDev/office-js-docs-pr/pull/778/commits/5eb63fee328426e5a647725d346edd3e3bd0e84f)_: 

:clock10: **Incremental building**: average incremental building time is _29 sec(s)_

**Comment by o365devx** [comment]
OPS Build status updates of commit _[5eb63fe](https://github.com/OfficeDev/office-js-docs-pr/pull/778/commits/5eb63fee328426e5a647725d346edd3e3bd0e84f)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/custom-functions-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/excel/custom-functions-best-practices.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-errors.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/excel/custom-functions-errors.md) | :white_check_mark:Succeeded | |
[docs/excel/custom-functions-overview.md](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/excel/custom-functions-overview.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/michelle-separateerrors/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C3%5C6%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201903062252554535-778%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=r04AkByQmGCLZ%2BHDhPlvTmSPiORl7uHID1taNH43oNA%3D&st=2019-03-06T22%3A49%3A23Z&se=2019-04-06T22%3A54%3A23Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/custom-functions-errors.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/excel/custom-functions-errors.md]: @keyur32 Can you return the built-in error types in Excel in custom functions? Just want to be sure we're documenting all the right stuff :)

[the remark this answers is not in the document]
Comment (mscharlock) [reply docs/excel/custom-functions-errors.md]: Ok confirmed that you can't do this until we build out this functionality

**Pull request merged**